src/index.js

```diff
@@ -1,117 +1,93 @@
-
-            argsList.push({
-                computeAsync: target => performComputation(target)
-            })
-
-            return performComputation()
-        }
-    })
-    if(autoRun) { proxy() }
-    return proxy
+const computed = (fun, {autoRun = true, callback = null} = {}) => {
+  const proxy = new Proxy(fun, {
+    apply (target, context, args) {
+      const performComputation = (fun = null) => {
+        computedStack.unshift(callback || proxy)
+        const result = fun ? fun() : target.apply(context, args)
+        computedStack.shift()
+        return result
+      }
+
+      args.push({computeAsync: performComputation})
+      return performComputation()
+    }
+  })
+  if (autoRun) proxy()
+  return proxy
 }
 
-const dispose = _ => _.__disposed = true
+const dispose = o => { o.__disposed = true }
 
-const batcher = {
-    timeout: null,
-    queue: new Set(),
-    process: () => {
-        for(let task of batcher.queue)
-            task()
-        batcher.queue.clear()
-        batcher.timeout = null
-    },
-    enqueue: task => {
-        if(batcher.timeout === null)
-            batcher.timeout = setTimeout(batcher.process, 0)
-        batcher.queue.add(task)
-    }
+const runAsync = async f => (await new Promise(resolve => resolve(f)))()
+
+const batcher = {
+  queue: new Set(),
+  process () {
+    each(batcher.queue, task => task()).clear()
+  },
+  enqueue (task) {
+    runAsync(batcher.process)
+    batcher.queue.add(task)
+  }
 }
 
-const observe = function(obj, options = {}) {
-    const { props = null, ignore = null, batch = false, deep = false } = options
-    observersMap.set(obj, new Map)
-
-    if(deep) {
-        if(obj instanceof Array) {
-            for(const key in obj) {
-                if(typeof obj[key] === 'object')
-                    obj[key] = observe(obj[key], options)
-            }
-        } else {
-            for(let key in obj) {
-                if(!obj.hasOwnProperty(key))
-                    continue
-                const value = obj[key]
-                if(typeof value === 'object') {
-                    obj[key] = observe(value, options)
-                }
-            }
-        }
-    }
+const observe = (obj, options = {}) => {
+  const {props = null, ignore = null, batch = false, deep = false} = options
+  observersMap.set(obj, new Map())
 
-    return new Proxy(obj, {
-        get(_, prop) {
-            const observerMap = observersMap.get(obj)
-
-            if((props && !props.includes(prop)) || (ignore && ignore.includes(prop)))
-                return obj[prop]
-
-            if(!observerMap.has(prop)) {
-                observerMap.set(prop, new Set())
-            }
-
-            const set = observerMap.get(prop)
-            if(computedStack.length > 0)
-                set.add(computedStack[0])
-            observerMap.set(prop, set)
-
-            return obj[prop]
-        },
-        set(_, prop, value) {
-            const observerMap = observersMap.get(obj)
```

Comment:
Moved this lower so that it's not in vain should one of the ``if`` statements return first.

Reply:
👍